Framework/DataObjects/src/PeaksWorkspace.cpp

```diff
@@ -149,6 +150,28 @@ void PeaksWorkspace::removePeak(const int peakNum) {
   peaks.erase(peaks.begin() + peakNum);
 }
 
+/** Removes multiple peaks
+* @param badPeaks peaks to be removed
+*/
+void PeaksWorkspace::removePeaks(std::vector<int> &badPeaks) {
+  if (badPeaks.size() == 0)
+    return;
+  auto first = peaks.begin();
+  auto end = peaks.end();
+  auto result = first;
+  std::sort(badPeaks.begin(), badPeaks.end());
```

Comment:
Sort then get your iterators - whilst the standard guarantees that not mutating the container's objects will preserve iterators it still looks fishy and can throw developers off if that makes sense?